testing/e2e/pkg/setup.go

```diff
@@ -0,0 +1,351 @@
+		ChainID:         testnet.Name,
+		GenesisTime:     time.Now().UTC(),
+		ConsensusParams: types.DefaultConsensusParams(),
+		AppState:        appState,
+		// AppHash is not provided but computed after InitChain
+	}, nil
+}
+
+func MakeConfig(node *Node) (*config.Config, error) {
+	cfg := config.DefaultConfig()
+	cfg.Moniker = node.Name
+	cfg.RPC.ListenAddress = "tcp://0.0.0.0:26657"
+	cfg.P2P.ExternalAddress = fmt.Sprintf("tcp://%v", node.AddressP2P(false))
+	cfg.P2P.PersistentPeers = strings.Join(node.Peers, ",")
+
+	// TODO: when we use adaptive timeouts, add a parameter in the testnet manifest
+	// to set block times
+	// FIXME: This values get overridden by the timeout consts in the app package.
+	// We should modify this if we want to quicken the time of the seconds.
+	cfg.Consensus.TimeoutPropose = 1000 * time.Millisecond
+	cfg.Consensus.TimeoutCommit = 300 * time.Millisecond
+	return cfg, nil
+}
+
+func InitNode(node *Node, genesis types.GenesisDoc, rootDir string) error {
+	// Initialize file directories
+	nodeDir := filepath.Join(rootDir, node.Name)
+	for _, dir := range []string{
+		filepath.Join(nodeDir, "config"),
+		filepath.Join(nodeDir, "data"),
+	} {
+		if err := os.MkdirAll(dir, os.ModePerm); err != nil {
+			return fmt.Errorf("error creating directory %s: %w", dir, err)
+		}
+	}
+
+	// Create and write the config file
+	cfg, err := MakeConfig(node)
+	if err != nil {
+		return fmt.Errorf("making config: %w", err)
+	}
+	config.WriteConfigFile(filepath.Join(nodeDir, "config", "config.toml"), cfg)
+
+	// Store the genesis file
+	err = genesis.SaveAs(filepath.Join(nodeDir, "config", "genesis.json"))
+	if err != nil {
+		return fmt.Errorf("saving genesis: %w", err)
+	}
+
+	// Create the app.toml file
+	appConfig, err := MakeAppConfig(node)
+	if err != nil {
+		return fmt.Errorf("making app config: %w", err)
+	}
+	serverconfig.WriteConfigFile(filepath.Join(nodeDir, "config", "app.toml"), appConfig)
+
+	// Store the node key for the p2p handshake
+	err = (&p2p.NodeKey{PrivKey: node.NetworkKey}).SaveAs(filepath.Join(nodeDir, "config", "node_key.json"))
+	if err != nil {
+		return err
+	}
+
+	// Store the validator signer key for consensus
+	(privval.NewFilePV(node.SignerKey,
+		filepath.Join(nodeDir, "config", "priv_validator_key.json"),
+		filepath.Join(nodeDir, "data", "priv_validator_state.json"),
+	)).Save()
+
+	return nil
+}
+
+func WriteDockerCompose(testnet *Testnet, file string) error {
+	// Must use version 2 Docker Compose format, to support IPv6.
+	tmpl, err := template.New("docker-compose").Parse(`version: '2.4'
+
+networks:
+  {{ .Name }}:
+    labels:
+      e2e: true
+    driver: bridge
+    ipam:
+      driver: default
+      config:
+      - subnet: {{ .IP }}
+
+services:
+{{- range .Nodes }}
+  {{ .Name }}:
+    labels:
+      e2e: true
+    container_name: {{ .Name }}
+    image: ghcr.io/celestiaorg/celestia-app:{{ index .Versions 0 }}
+    entrypoint: ["/bin/celestia-appd"]
+    command: ["start"]
+    init: true
```

Comment:
my local docker setup might be wonk, but I had to add 
`    user: root` in order to get the test to work. If not, the test will appear to be waiting, but none of the nodes actually start because they do not have permission to write to the volume.